src/storage/v2/disk/storage.cpp

```diff
@@ -1469,6 +1469,10 @@ std::vector<EdgeAccessor> DiskStorage::OutEdges(const VertexAccessor *src_vertex
   std::vector<EdgeAccessor> result;
   auto out_edges = utils::Split(out_edges_str, ",");
   for (const std::string &edge_gid_str : out_edges) {
+    if (hops_limit && hops_limit->has_value()) {
+      --(**hops_limit);
```

Comment:
would make a method that decrements the counter. and another to see if the hops is exhausted